test/functional/p2p_1p1c_package_relay.py

```diff
@@ -0,0 +1,223 @@
+        with node.assert_debug_log([f"got inv: wtx {low_fee_child['wtxid']}  have"]):
+            peer2.send_and_ping(msg_inv([CInv(t=MSG_WTX, h=low_child_wtxid_int)]))
+        assert "getdata" not in peer2.last_message
+        # If either peer sends the parent again, package evaluation should not be attempted
+        with node.assert_debug_log([f"didn't evaluate package for {low_fee_parent['txid']} (wtxid={low_fee_parent['wtxid']}), no eligible child found"]):
+            peer1.send_and_ping(msg_tx(low_fee_parent["tx"]))
+        with node.assert_debug_log([f"didn't evaluate package for {low_fee_parent['txid']} (wtxid={low_fee_parent['wtxid']}), no eligible child found"]):
+            peer2.send_and_ping(msg_tx(low_fee_parent["tx"]))
+
+        self.log.info("Check that the node groups a low-feerate tx with its single child in orphanage")
+        # Send the high feerate (orphan) child
+        high_child_wtxid_int = int(high_fee_child["tx"].getwtxid(), 16)
+        peer1.send_and_ping(msg_inv([CInv(t=MSG_WTX, h=high_child_wtxid_int)]))
+        peer1.wait_for_getdata([high_child_wtxid_int])
+        peer1.send_and_ping(msg_tx(high_fee_child["tx"]))
+        # Node should request the orphan's parent, even though it has already been rejected, both by
+        # itself and with a child. This is necessary, otherwise high_fee_child can be censored.
+        parent_txid_int = int(low_fee_parent["txid"], 16)
+        peer1.wait_for_getdata([parent_txid_int])
+        expected_logs_1p1c = []
+        expected_logs_1p1c.append(f"found tx {low_fee_parent['txid']} (wtxid={low_fee_parent['wtxid']}) in reconsiderable rejects, looking for child in orphanage")
+        expected_logs_1p1c.append(f"found child {high_fee_child['txid']} (wtxid={high_fee_child['wtxid']}) of tx {low_fee_parent['txid']} (wtxid={low_fee_parent['wtxid']}) in orphanage, trying package evaluation")
+        with node.assert_debug_log(expected_logs_1p1c):
+            peer1.send_and_ping(msg_tx(low_fee_parent["tx"]))
+
+        # Both transactions should now be in mempool
+        node_mempool = node.getrawmempool()
+        assert low_fee_parent["txid"] in node_mempool
+        assert high_fee_child["txid"] in node_mempool
+
+        peer1.peer_disconnect()
+        peer2.peer_disconnect()
+        self.sync_all()
+
+    def run_test(self):
+        self.ctr = 0
+        self.wallet = MiniWallet(self.nodes[1])
+        self.generate(self.wallet, 120)
+
+        self.log.info("Fill mempools with large transactions to raise mempool minimum feerates")
+        self.raise_network_minfee()
+
+        self.log.info("Check 1p1c validation logic on a single node")
+        self.test_individual_logic()
+
+        self.log.info("Check end-to-end package relay across multiple nodes")
+        self.packages_to_submit = []
+        self.transactions_to_presend = [[]] * self.num_nodes
+        self.replacement_packages = []
```

Comment:
this is never set

Reply:
Oops, forgot to delete from a package RBF test I built on top. Fixed